### PR TITLE
build: Add pre-commit hook to validate templates

### PR DESCRIPTION
**Pull request by jimmidyson**

### CODE REVIEW

**Comment by jimmidyson** [comment]
retest this please

File: support/validate-generated-templates.sh
Comment (rhuss) [inline support/validate-generated-templates.sh]: OS X's `mktemp` doesn't support `-p` (what does it anyway ? ;-)

File: support/validate-generated-templates.sh
Comment (rhuss) [inline support/validate-generated-templates.sh]: Parent dir of script ?

**Review by rhuss — DISMISSED** [review]
cool, thanks ! a minor comment wrt OS X compatibility

File: support/validate-generated-templates.sh
Comment (jimmidyson) [reply support/validate-generated-templates.sh]: `-p` specifies the directory to create the temporary directory in. This was basically saying create it in the same dir as source dir but can drop that requirement, no worries.

**Comment by jimmidyson** [comment]
This is ready for review before merge - has been a good intro to Jenkins pipelines and what is possible, more stuff to come now I hope to make Syndesis CI even smoother.

File: support/validate-generated-templates.sh
Comment (rhuss) [inline support/validate-generated-templates.sh]: Has to be ${DIR}/generator/run.sh

File: support/validate-generated-templates.sh
Comment (rhuss) [inline support/validate-generated-templates.sh:8]: thx, works now on OS X

**Review by rhuss — DISMISSED** [review]
Can't really comment on the Jenkinsfile, so if there are no objections I approve this PR nevertheless (could be reverted or fixed then later)

**Review by rhuss — COMMENTED** [review]
Just verified that pre-commit script works on OS X, too.

Just this minor fix (see above), will approve immediately after this.

File: support/validate-generated-templates.sh
Comment (jimmidyson) [reply support/validate-generated-templates.sh]: Thanks!

**Comment by jimmidyson** [comment]
Updated.

**Comment by jimmidyson** [comment]
retest this please

File: CONTRIBUTING.md
Comment (rhuss) [inline CONTRIBUTING.md:13]: One final remark: I would add `--allow-missing-config` as option here, otherwise any commit without a `.pre-commit-config.yaml` will fail (e.g. in other git repos)

**Review by rhuss — APPROVED** [review]

File: CONTRIBUTING.md
Comment (rhuss) [reply CONTRIBUTING.md:13]: I'm going to update the readme separately if you agree to this.

Comment (jimmidyson) [reply CONTRIBUTING.md:13]: This shouldn't have any effect in other repos, it needs to be run in every repo you want to run pre commit in, so I don't think this is necessary.

Comment (rhuss) [reply CONTRIBUTING.md:13]: ah, got it. The reason why I had problem is that I run it on this branch, but then switched back to master (but the hook has already installed in `.git/`).

**Pull request merged**
